tests/s2n_test.h

```diff
@@ -48,6 +56,13 @@ int test_count;
     EXPECT_SUCCESS_WITHOUT_COUNT(s2n_in_unit_test_set(true));  \
     S2N_TEST_OPTIONALLY_ENABLE_FIPS_MODE();                    \
     EXPECT_SUCCESS_WITHOUT_COUNT(s2n_init());                  \
+    if (s2n_pq_is_enabled()) {                                 \
+        TEST_ALL = "test_all";                                 \
+        TEST_ALL_TLS12 = "test_all_tls12";                     \
+    } else {                                                   \
+        TEST_ALL = "test_all_no_pq";                           \
+        TEST_ALL_TLS12 = "test_all_tls12_no_pq";               \
+    }                                                          \
```

Comment:
I've been going back and forth on this, but I think I've settled on disagreeing with this approach.

Our current behavior for security policies is that we ignore any settings that don't work with the current build. If our libcrypto doesn't support a signature scheme, we ignore that signature scheme. If fips doesn't allow a cipher suite, we skip that cipher suite. The same should be true of PQ: if our build doesn't support it, ignore it.

If we start requiring separate pq / non-pq security policies, are we going to do the same for other settings? Like RSA-PSS / non-RSA-PSS security policies? That's potentially going to make it hard for customers to manage their security policies.

Reply:
Agreed. I've updated this PR to abandon the approach of enforcing a non-pq security policy selection if PQ is disabled, and remove the test non-pq security policies. That prunes the PR of a lot of code changes; what's left is fairly minimal:

* the run-time function to check if pq is enabled
* small rewrite of the set_cipher_pref functions to perform the null checks before assignemnt
* improvements to the pq intev2 tests
* straightforward unit test improvements